x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -505,18 +505,20 @@ export class KnowledgeBaseService {
     try {
       const response = await this.dependencies.esClient.asInternalUser.search<KnowledgeBaseEntry>({
         index: resourceNames.aliases.kb,
-        ...(query
-          ? {
-              query: {
-                wildcard: {
-                  doc_id: {
-                    value: `${query}*`,
-                  },
-                },
-              },
-            }
-          : {}),
+        query: {
+          bool: {
+            // filter by search query
+            filter: [
+              ...(query ? [{ wildcard: { doc_id: { value: `${query}*` } } }] : [{ match_all: {} }]),
+            ],
+
+            // boost score of user instruction entries
+            should: [{ term: { type: KnowledgeBaseType.UserInstruction } }],
+            minimum_should_match: 0,
+          },
```

Comment:
TODO: filter out user instructions here, and instead only load them (it) when opening the flyout for editing the user instruction.
Also: Add an API test to ensure that only the user instruction belonging to the user is retrieved when opening the flyout